packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -1005,6 +1142,34 @@ class MediaQuery extends InheritedModel<_MediaQueryAspect> {
   /// Creates a widget that provides [MediaQueryData] to its descendants.
   const MediaQuery({super.key, required this.data, required super.child});
 
+  /// Creates a new [MediaQuery] that inherits from the ambient [MediaQuery]
+  /// from the given context, but applies the specified text style overrides.
+  ///
+  /// The [context] argument must have a [MediaQuery] in scope.
+  ///
+  /// If a text style override argument is null (the default), then the
+  /// corresponding override in the returned [MediaQueryData] is set to null.
+  ///
+  /// See also:
+  ///
+  ///  * [MediaQueryData.lineHeightScaleFactorOverride], [MediaQueryData.letterSpacingOverride],
+  ///    [MediaQueryData.wordSpacingOverride], [MediaQueryData.paragraphSpacingOverride], the
+  ///    affected properties of the [MediaQueryData].
+  MediaQuery.applyTextStyleOverrides({
+    super.key,
+    required BuildContext context,
+    required double? lineHeightScaleFactorOverride,
+    required double? letterSpacingOverride,
+    required double? wordSpacingOverride,
+    required double? paragraphSpacingOverride,
+    required super.child,
+  }) : data = MediaQuery.of(context).applyTextStyleOverrides(
```

Comment:
This looks error-prone. `data` is evaluated on construction so if the user caches this widget this doesn't get rebuilt even if the ancestor media query changes.

Reply:
I'd just wrap this in a builder, and you don't have to take `context` if you do that.

Reply:
Or move this to `MediaQueryData`.

Reply:
Oh good catch. +1 to wrapping this in a builder and removing the `context` argument.